### ISSUE TITLE
Exceptions issues when driver connects to unreachable ArangoDB

### ISSUE THREAD

**Issue by suimp** [arangodb/arangodb-java-driver#270]
Hello, I have encountered a specific exceptions throws while was trying to hadle retries (in case when connection with DB is lost) to ArangoDB using java Driver.

If I'm not mistaken, in the previous versions of java driver there was ArangoConfigure class, where you can handle retries (setConnectRetryCount and setConnectRetryWait)
but in 4+ driver version there are no such options in ArangoDB.Builder.

Driver 5.0.2 and the following arangoDB configuration are used:
```
ArangoDB.Builder()
.host(someHostname, somePort)
.user(someLogin)
.password(somePassword)              
.maxConnections(someConnectionsNumber)
.acquireHostList(true);
```

When I try to get some data from arangoDB(execute any method of _ArangoDatabase_ class) while arangoDB is unreachable, I receive these exceptions going cyclically: _ConnectException_ and _ArangoDBException: Was not able to connect to any host_ 

**!look at the attached file**

The cases I faced:

1.
- The application and arangoDB are started and ready for work
- Then I shut down arangoDB
- Try to get something from arangoDB (using the application)
- Get cyclically going exceptions (as I described above)

2.
- The application and arangoDB are started and ready for work
- Then I shut down arangoDB
- Try to get something from arangoDB (using the application)
- **!** On some iteration I receive the _ConnectException_, after that I up the arango
- Try to get something from arangoDB (using the application) and as a result - I receive _ArangoDBException: Was not able to connect to any host_ 

3.
Sometimes the other situation happens:
After the shutting down of arangoDB, I call endpoint to arangoDB and driver tries to connect to host again and again:
```
2019-06-14 16:22:06.556  WARN [-,602a0e028d985e7b,602a0e028d985e7b,false] 12460 --- [qtp988637661-46] c.a.i.velocystream.VstCommunication      : Could not connect to host[addr=127.0.0.1,port=6002] or SSL Handshake failed. Try connecting to host[addr=127.0.0.1,port=6002]
2019-06-14 16:22:07.557  WARN [-,602a0e028d985e7b,602a0e028d985e7b,false] 12460 --- [qtp988637661-46] c.a.i.velocystream.VstCommunication      : Could not connect to host[addr=127.0.0.1,port=6002] or SSL Handshake failed. Try connecting to host[addr=127.0.0.1,port=6002]
2019-06-14 16:22:08.559  WARN [-,602a0e028d985e7b,602a0e028d985e7b,false] 12460 --- [qtp988637661-46] c.a.i.velocystream.VstCommunication      : Could not connect to host[addr=127.0.0.1,port=6002] or SSL Handshake failed. Try connecting to host[addr=127.0.0.1,port=6002]
2019-06-14 16:22:09.562  WARN [-,602a0e028d985e7b,602a0e028d985e7b,false] 12460 --- [qtp988637661-46] c.a.i.velocystream.VstCommunication      : Could not connect to host[addr=127.0.0.1,port=6002] or SSL Handshake failed. Try connecting to host[addr=127.0.0.1,port=6002]
```
after the arangoDB is up the follwing exception is thown:

```
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is com.arangodb.ArangoDBException: Response Code: 503
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:982)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:866)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:851)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
```

Because of such behavior the only way it is possible to do retries in iteration the _ArangoDBException: Was not able to connect to any host_ 
 is thrown is: to check the message of this general exception (it doesn't have any reasonable cause) 

**!pay attention:**

1. To make things clear: "the first endpoint call returns the ConnectException, the second endpoint call returns the "ArangoDBException: Was not able to connect to any host", the 3d call returns the ConnectException, the 4th -"ArangoDBException: Was not able to connect to any host"... asf

2. ConnectException - example of exception which not always rapped as arangoDB one (see attached file)

3. package: com.arangodb.internal.velocystream;
class: VstCommunication
method: connect
block:
```
if (host == null) {
                hostHandler.reset();
                throw new ArangoDBException("Was not able to connect to any host");
            }
```

When the driver fails to connect (_VstCommunication_ -> _connect_ -> _catch_ block -> _hostHandler.fail()_) more then 3 times in a row, the rethrow is happened (throw new ArangoDBException(e)) and host is becoming equal to null and since we in the endless "_while_", in the next iteration,  we pass the first null checkpoint for host and the general ArangoException is thrown without any appropriate cause(it has only the message = "Was not able to connect to any host")

4. Method "try to get something from ArangoDB" is ArangoDatabase.query()

In the issue I expect this exceptions problem is resolved (so it can easy to do retries)
And, since earlier you had a preset for retries, also it would be great these configuration options are
 come back. 

**_Attached file:_** [logs.2.0.txt](https://github.com/arangodb/arangodb-java-driver/files/3291249/logs.2.0.txt)

Thank you!

**Comment by rashtao** [comment]
@suimp this should be fixed by https://github.com/arangodb/arangodb-java-driver/commit/1b66cac5e1a1596767bce6acd69a9dd7ea165bca?w=1

**Comment by rashtao** [comment]
fixed in v6.3.0
